src/dash-table/dash/DataTable.js

```diff
@@ -434,6 +441,26 @@ export const propTypes = {
      */
     editable: PropTypes.bool,
 
+    /**
+     * If True, then the columns will be hideable through the UI.
+     * When `hideable` is True, particular columns can be made
+     * un-hideable by setting `hideable` to `False` in the `columns`
+     * property.
+     *
+     * When `hideable` is False, particular columns can be made
+     * hideable by setting `hideable` to `True` in the `columns`
+     * property.
+     */
+    hideable: PropTypes.bool,
+
+    /**
+     * 'beta'. Define the header row on which the hideable action will be
+     * visible.
+     *
+     * Defaults to the last header row.
+     */
+    hideable_row: PropTypes.number,
```

Comment:
why not continue the pattern from `clearable`, `deletable`, and `renamable` of supporting an array of bools?

Reply:
The main issue I've been having is with the reverse operation (toggling the column back into the table) -- these other operations do not have this reverse.

With the toggle menu, I find that the menu would be messy if it had to show all combinations / options of merged cells based on various header rows that could be used to toggle them.

Instead, this prop forces the `hide` action to exist on a single header row. Making it possible to show items in the toggle menu that are all at the same level.

I'm aware that this prop may not represent the best solution in the long termand that's why I've taken the liberty to tag it as `beta`.

Reply:
also: mention these new props in the changelog

Reply:
Personally I don't see exact parity between the two toggle mechanisms as being important - or perhaps even desirable. I feel like the menu should list all columns and allow toggling each independently, even if the hideable button in the header itself is placed in a merged cell and hides all those columns together.

Also, not sure if we had this discussion before or not, but seems to me like a single `true` in any of these props would be better as "just the last row", and if someone really wants all rows to get the buttons they would need to provide `[true, true, true]`. Is that too much of a breaking change to discuss making now?

Reply:
No no. Let's discuss this.

Reply:
OK, after offline discussion for context, we agreed (I think, correct me if I'm missing a piece @Marc-Andre-Rivet):
- `hideable` can be a bool or array of bools, plus (and all similar props get this option too) `'last'` to only apply to the last header row. That way it's easy to get this behavior without needing to create an array `[false, false, false, true]` but it won't break the existing `true` that turns on all rows.
- The hide/show menu gets a list corresponding to all possible hideable buttons. Each entry is the concatenation of all header rows above and including the one the button is in, ordered by column index then row index. This may look weird if you've shown buttons in multiple rows, but we're OK with that because the behavior is weird in that case too, and the user was explicit about enabling this weirdness.
- The behavior of this menu follows the button on screen: ie if you have a merged header and *any* of the merged columns is visible (even if you've hidden some of them using a lower down hide button) then the checkbox is checked, and unchecking it hides all remaining columns. After that, checking the box to re-show will reshow all merged columns, it will not attempt to invert to the partially hidden state.